File: bukkit/elysium-chat-bukkit/src/main/java/com/seventh_root/elysium/chat/bukkit/command/chatchannel/ChatChannelCommand.java
package com.seventh_root.elysium.chat.bukkit.command.chatchannel;

import com.seventh_root.elysium.chat.bukkit.ElysiumChatBukkit;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;

public class ChatChannelCommand implements CommandExecutor {

    private final ElysiumChatBukkit plugin;

    private final ChatChannelJoinCommand chatChannelJoinCommand;
    private final ChatChannelLeaveCommand chatChannelLeaveCommand;
    private final ChatChannelSpeakCommand chatChannelSpeakCommand;
    private final ChatChannelCreateCommand chatChannelCreateCommand;
    private final ChatChannelDeleteCommand chatChannelDeleteCommand;
    private final ChatChannelListCommand chatChannelListCommand;
    private final ChatChannelSetCommand chatChannelSetCommand;

    public ChatChannelCommand(ElysiumChatBukkit plugin) {
        this.plugin = plugin;
        chatChannelJoinCommand = new ChatChannelJoinCommand(plugin);
        chatChannelLeaveCommand = new ChatChannelLeaveCommand(plugin);
        chatChannelSpeakCommand = new ChatChannelSpeakCommand(plugin);
        chatChannelCreateCommand = new ChatChannelCreateCommand(plugin);
        chatChannelDeleteCommand = new ChatChannelDeleteCommand(plugin);
        chatChannelListCommand = new ChatChannelListCommand(plugin);
        chatChannelSetCommand = new ChatChannelSetCommand(plugin);
    }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if (args.length > 0) {
            String[] newArgs = new String[args.length - 1];
            System.arraycopy(args, 1, newArgs, 0, args.length - 1);
            if (args[0].equalsIgnoreCase("join") || args[0].equalsIgnoreCase("unmute")) {
                return chatChannelJoinCommand.onCommand(sender, command, label, newArgs);
            } else if (args[0].equalsIgnoreCase("leave") || args[0].equalsIgnoreCase("mute")) {
                return chatChannelLeaveCommand.onCommand(sender, command, label, newArgs);
            } else if (args[0].equalsIgnoreCase("speak") || args[0].equalsIgnoreCase("talk")) {
                return chatChannelSpeakCommand.onCommand(sender, command, label, newArgs);
            } else if (args[0].equalsIgnoreCase("create") || args[0].equalsIgnoreCase("new") || args[0].equalsIgnoreCase("add")) {
                return chatChannelCreateCommand.onCommand(sender, command, label, newArgs);
            } else if (args[0].equalsIgnoreCase("delete") || args[0].equalsIgnoreCase("remove")) {
                return chatChannelDeleteCommand.onCommand(sender, command, label, newArgs);
            } else if (args[0].equalsIgnoreCase("list")) {
                return chatChannelListCommand.onCommand(sender, command, label, newArgs);
            } else if (args[0].equalsIgnoreCase("set") || args[0].equalsIgnoreCase("modify")) {
                return chatChannelSetCommand.onCommand(sender, command, label, newArgs);
            } else {
                sender.sendMessage(ChatColor.translateAlternateColorCodes('&', plugin.getConfig().getString("messages.chatchannel-usage")));
            }
        } else {
            sender.sendMessage(ChatColor.translateAlternateColorCodes('&', plugin.getConfig().getString("messages.chatchannel-usage")));
        }
        return true;
    }

}

File: bukkit/elysium-chat-bukkit/src/main/kotlin/com/seventh_root/elysium/chat/bukkit/command/chatchannel/ChatChannelCommand.kt
/*
 * Copyright 2016 Ross Binden
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.seventh_root.elysium.chat.bukkit.command.chatchannel

import com.seventh_root.elysium.chat.bukkit.ElysiumChatBukkit
import com.seventh_root.elysium.chat.bukkit.chatchannel.ElysiumChatChannelProvider
import com.seventh_root.elysium.players.bukkit.player.ElysiumPlayerProvider
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

/**
 * Chat channel command.
 * Sets which chat channel a player is speaking in.
 */
class ChatChannelCommand(private val plugin: ElysiumChatBukkit): CommandExecutor {

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
        if (sender is Player) {
            if (args.size > 0) {
                val playerProvider = plugin.core.serviceManager.getServiceProvider(ElysiumPlayerProvider::class)
                val chatChannelProvider = plugin.core.serviceManager.getServiceProvider(ElysiumChatChannelProvider::class)
                val player = playerProvider.getPlayer(sender)
                val chatChannel = chatChannelProvider.getChatChannel(args[0])
                if (chatChannel != null) {
                    if (sender.hasPermission("elysium.chat.command.chatchannel.${chatChannel.name}")) {
                        chatChannel.addSpeaker(player)
                        sender.sendMessage(ChatColor.translateAlternateColorCodes('&', plugin.config.getString("messages.chatchannel-valid"))
                                .replace("\$channel", chatChannel.name))
                    } else {
                        sender.sendMessage(ChatColor.translateAlternateColorCodes('&', plugin.config.getString("messages.no-permission-chatchannel"))
                                .replace("\$channel", chatChannel.name))
                    }
                } else {
                    sender.sendMessage(ChatColor.translateAlternateColorCodes('&', plugin.config.getString("messages.chatchannel-invalid-chatchannel")))
                }
            } else {
                sender.sendMessage(ChatColor.translateAlternateColorCodes('&', plugin.config.getString("messages.chatchannel-usage")))
            }
        } else {
            sender.sendMessage(ChatColor.translateAlternateColorCodes('&', plugin.config.getString("messages.not-from-console")))
        }
        return true
    }

}